### PR TITLE
Remove the Flask app from the CLI to speed it up

### PR DESCRIPTION
**Pull request by joanise**
This is a pretty significant refactoring effort to move all expensive imports out of two places:
 - `g2p/__init__.py`: just importing `g2p` should not trigger loading and importing everything
 - `g2p/cli.py`: just calling `g2p -h` or `g2p cmd -h` should not trigger loading and importing everything

In both cases, the imports are tucked away inside the functions that need them.

However, we have constants and interfaces we need higher up, so I moved those to a new `g2p/constants.py` file. It's not super elegant, but I think it's at least a partial design improvement: instead of making a top-level module need a deeper one just for its constants or its class definition for typing, I moved them into a lightweight file that both import. This is definitely up for discussion and potential improvements.

But... `g2p -h` now takes just 0.9 seconds on my machine, and a mere 0.05 seconds in the CI run that just finished a few minutes ago.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -46,6 +46,28 @@ jobs:
...
+        run: |
+          PYTHONPROFILEIMPORTTIME=1 g2p -h 2> importtime.txt > /dev/null
+          CLI_LOAD_TIME="$((/usr/bin/time --format=%E g2p -h > /dev/null) 2>&1)"
+          echo "CLI load time: $CLI_LOAD_TIME"
+          cat importtime.txt
+          echo "CLI load time: $CLI_LOAD_TIME" > import-message.txt
+          echo "PR head ${{ github.event.pull_request.head.sha }}" >> import-message.txt
+          echo "Imports that take more than 0.1 s:" >> import-message.txt
+          grep -E 'cumulative|[0-9]{6}' importtime.txt >> import-message.txt
+          if [[ "$CLI_LOAD_TIME" > "0:01.00" ]]; then \
```
Comment (dhdaines) [inline .github/workflows/tests.yml:60]: wow, does this work?

File: g2p/constants.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+LANGS_FILE_NAME = "langs.json.gz"
+NETWORK_FILE_NAME = "network.json.gz"
+
+
+class BaseTransducer(ABC):
+    """Base class to typecheck transducers without having to import them."""
+
+    @abstractmethod
+    def __call__(self, to_convert: str):
+        pass
```
Comment (dhdaines) [inline g2p/constants.py]: You could add a `#pragma: no cover` here,  I suppose, or just a bogus test that calls this abstract method and ensures that it does nothing and returns `None`.

Likewise below (I'm not going to add another comment just to say the same thing)

File: g2p/constants.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    "fast_levenshtein_distance",
+    "levenshtein_distance",
+]
+
+LANGS_DIR = os.path.join(os.path.dirname(__file__), "mappings", "langs")
+LANGS_FILE_NAME = "langs.json.gz"
+NETWORK_FILE_NAME = "network.json.gz"
+
+
+class BaseTransducer(ABC):
```
Comment (dhdaines) [inline g2p/constants.py:21]: I mean strictly speaking these are not *constants*, so you could decide if it's worth the trouble to put them in another separate module called `g2p.types` or `g2p.abc`.

**Review by dhdaines — APPROVED** [review]
Looks good to me, comments are optional changes.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (joanise) [reply .github/workflows/tests.yml:60]: Yeah, it's sweet. We've been using this in EveryVoice, that's where I copied it from.

File: g2p/constants.py (hunk 3 above)
Comment (joanise) [reply g2p/constants.py:21]: That's a good point. I didn't like the abuse of the name, `g2p.types` would be a lot clearer.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (joanise) [reply .github/workflows/tests.yml:60]: OK, it's a bit clunky, but in CI I don't mind that too much, and the message you get in the PR conversation is really nice.

As for `"$CLI_LOAD_TIME" > "0:01.00"` doing the right thing, I force the time output to that syntax with `/usr/bin/time --format=%E` higher up.

File: g2p/constants.py (hunk 3 above)
Comment (joanise) [reply g2p/constants.py]: Refactored now.

**Review by roedoejet — APPROVED** [review]
Thanks for doing this @joanise ! I tested this locally and `g2p -h` took 0.076s 🚀
